dolphin/api/v1/storages.py

```diff
@@ -135,8 +134,35 @@ def update(self, req, id, body):
     def delete(self, req, id):
         return webob.Response(status_int=http_client.ACCEPTED)
 
+    @wsgi.response(202)
     def sync_all(self, req):
-        return dict(name="Sync all storages")
+        """
+        :param req:
+        :return: it's a Asynchronous call. so return 202 on success. sync_all
+        api performs the storage device info, pool, volume etc. tasks on each
+        registered storage device.
+        """
+        # validate the id
+        ctxt = req.environ['dolphin.context']
+        try:
+            storages = db.storage_get_all(ctxt)
+            if not storages:
+                msg = _("No registered storage found in database")
```

Comment:
I think . can be added at end of msg for consistency

Reply:
ok. I will update this with others comment

Reply:
there is no need to judege whether storages are empty or not, just keep for loop:
```
            for storage in storages:
                for subclass in task.StorageResourceTask.__subclasses__():
                    self.task_rpcapi.sync_storage_resource(
                        ctxt,
                        storage['id'],
                        subclass.__module__ + '.' + subclass.__name__
                    )
```